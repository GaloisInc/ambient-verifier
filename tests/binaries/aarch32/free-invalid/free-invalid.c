#include <stdlib.h>

int main(void) {
  // Double free
  int *x = malloc(sizeof(int));
  free(x);
  free(x);

  // Use after free
  int *y = malloc(sizeof(int));
  *y = 42;
  free(y);
  *y = 27;

  // free on pointer that doesn't point to the heap
  int wval = 42;
  int *w = &wval;
  free(w);

  return 0;
}
