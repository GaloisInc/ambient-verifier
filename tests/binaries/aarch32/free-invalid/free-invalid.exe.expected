successful: 0
failed: 6
